src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -39,45 +42,46 @@ public static IList<INamedTypeSymbol> GetAllInterfacesIncludingThis(this ITypeSy
             return allInterfaces;
         }
 
-        public static bool IsAbstractClass(this ITypeSymbol symbol)
+        public static bool IsAbstractClass([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.TypeKind == TypeKind.Class && symbol.IsAbstract;
         }
 
-        public static bool IsSystemVoid(this ITypeSymbol symbol)
+        public static bool IsSystemVoid([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.SpecialType == SpecialType.System_Void;
         }
 
-        public static bool IsNullable(this ITypeSymbol symbol)
+        public static bool IsNullable([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
             => symbol?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
 
-        public static bool IsModuleType(this ITypeSymbol symbol)
+        public static bool IsModuleType([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.TypeKind == TypeKind.Module;
         }
 
-        public static bool IsInterfaceType(this ITypeSymbol symbol)
+        public static bool IsInterfaceType([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.TypeKind == TypeKind.Interface;
         }
 
-        public static bool IsDelegateType(this ITypeSymbol symbol)
+        public static bool IsDelegateType([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.TypeKind == TypeKind.Delegate;
         }
 
-        public static bool IsStructType(this ITypeSymbol symbol)
+        public static bool IsStructType([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
         {
             return symbol?.TypeKind == TypeKind.Struct;
         }
 
-        public static bool IsAnonymousType(this INamedTypeSymbol symbol)
+        public static bool IsAnonymousType([NotNullWhen(returnValue: true)] this INamedTypeSymbol? symbol)
         {
             return symbol?.IsAnonymousType == true;
         }
 
-        public static ITypeSymbol RemoveNullableIfPresent(this ITypeSymbol symbol)
+        [return: NotNullIfNotNull(parameterName: "symbol")]
+        public static ITypeSymbol? RemoveNullableIfPresent([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
```

Comment:
The `NotNullWhen(returnValue: true)` looks fishy here.

Reply:
it does